go/ct/rlz/conditions.go

```diff
@@ -962,3 +985,77 @@ func (c *hasNoBlobHash) String() string {
 }
 
 ////////////////////////////////////////////////////////////
+// Is read only mode
+
+type isReadOnly struct {
+}
+
+func IsReadOnly() Condition {
+	return &isReadOnly{}
+}
+
+func (c *isReadOnly) Check(s *st.State) (bool, error) {
+	return s.ReadOnly, nil
+}
+
+func (c *isReadOnly) Restrict(generator *gen.StateGenerator) {
+	generator.SetReadOnly(true)
+}
+
+func (c *isReadOnly) GetTestValues() []TestValue {
+	property := Property(c.String())
+	domain := boolDomain{}
+	restrict := func(generator *gen.StateGenerator, isReadOnly bool) {
+		if isReadOnly {
+			generator.SetReadOnly(true)
+		} else {
+			generator.SetReadOnly(false)
+		}
+	}
+	return []TestValue{
+		NewTestValue(property, domain, true, restrict),
+	}
+}
+
+func (c *isReadOnly) String() string {
+	return "isReadOnly()"
+}
+
+////////////////////////////////////////////////////////////
+// Is not read only mode
+
+type isNotReadOnly struct {
+}
+
+func IsNotReadOnly() Condition {
+	return &isNotReadOnly{}
+}
+
+func (c *isNotReadOnly) Check(s *st.State) (bool, error) {
+	return !s.ReadOnly, nil
+}
+
+func (c *isNotReadOnly) Restrict(generator *gen.StateGenerator) {
+	generator.SetReadOnly(false)
+}
+
+func (c *isNotReadOnly) GetTestValues() []TestValue {
+	property := Property(c.String())
```

Comment:
Same here, the positive (`IsReadOnly`) and negative (`IsNotReadOnly`) conditions should reference the same `property`. Also, please avoid code duplication by, for instance, providing a stand-alone version of this function accepting a `[]bool` listing all the `TestValue`s that should be used.